core/node/da_clients/src/eigen/memstore.rs

```diff
@@ -0,0 +1,232 @@
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+    time::{Duration, Instant},
+};
+
+use anyhow::Error;
+use rand::{rngs::OsRng, Rng, RngCore};
+use sha3::{Digest, Keccak256};
+use tokio::time::interval;
+use zksync_config::configs::da_client::eigen::MemStoreConfig;
+use zksync_da_client::types::{DAError, DispatchResponse, InclusionData};
+
+use super::{
+    blob_info::{self, BlobInfo},
+    client::to_retriable_error,
+};
+
+#[derive(Debug, PartialEq)]
+pub enum MemStoreError {
+    BlobToLarge,
+    BlobAlreadyExists,
+    IncorrectCommitment,
+    #[cfg(test)]
+    BlobNotFound,
+}
+
+impl Into<Error> for MemStoreError {
+    fn into(self) -> Error {
+        match self {
+            MemStoreError::BlobToLarge => Error::msg("Blob too large"),
+            MemStoreError::BlobAlreadyExists => Error::msg("Blob already exists"),
+            MemStoreError::IncorrectCommitment => Error::msg("Incorrect commitment"),
+            #[cfg(test)]
+            MemStoreError::BlobNotFound => Error::msg("Blob not found"),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct MemStoreData {
+    store: HashMap<String, Vec<u8>>,
+    key_starts: HashMap<String, Instant>,
+}
+
+#[derive(Clone, Debug)]
+pub struct MemStore {
+    config: MemStoreConfig,
+    data: Arc<RwLock<MemStoreData>>,
+}
+
+impl MemStore {
+    pub fn new(config: MemStoreConfig) -> Arc<Self> {
+        let memstore = Arc::new(Self {
+            config,
+            data: Arc::new(RwLock::new(MemStoreData {
+                store: HashMap::new(),
+                key_starts: HashMap::new(),
+            })),
+        });
+        let store_clone = Arc::clone(&memstore);
+        tokio::spawn(async move {
+            store_clone.pruning_loop().await;
+        });
+        memstore
+    }
+
+    async fn put_blob(self: Arc<Self>, value: Vec<u8>) -> Result<Vec<u8>, MemStoreError> {
+        tokio::time::sleep(Duration::from_millis(self.config.put_latency)).await;
+        if value.len() as u64 > self.config.max_blob_size_bytes {
+            return Err(MemStoreError::BlobToLarge.into());
+        }
+
+        // todo: Encode blob?
```

Comment:
Is this encoding mentioned the one it's done in line 120?

```rs
let cert_bytes = rlp::encode(&blob_info).to_vec();
```

Reply:
It was related to proxy codec, we wont be using it